pg_backup_api/pg_backup_api/spec/pg_backup_api.yaml

```diff
@@ -1,21 +1,869 @@
+          type: string
+        pgespresso_installed:
+          description: Whether the PostgreSQL server has Pgespresso installed
+          type: boolean
+        postgresql_systemid:
+          description: The system identifier of the PostgreSQL server
+          type: string
+        replication_slot:
+          $ref: "#/components/schemas/ReplicationSlot"
+        replication_slot_support:
+          description: Whether the PostgreSQL server supports replication slots
+          type: boolean
+        server_txt_version:
+          description: The version of the PostgreSQL server as human-readable text
+          type: string
+        stats_reset:
+          description: >
+            The time at which statistics were last reset for this PostgreSQL server
+            (in ctime format in the local timezone of the server)
+          type: string
+        synchronous_standby_names:
+          description: >
+            The current value of synchronous_standby_names on the PostgreSQL server
+          type: array
+          items:
+            description: >
+              Name of a standby server which can support synchronous replication
+            type: string
+        wal_compression:
+          description: The current value of wal_compression on the PostgreSQL server
+          type: string
+        wal_keep_size:
+          description: The current value of wal_keep_size on the PostgreSQL server
+          type: string
+        wal_level:
+          description: The current value of wal_level on the PostgreSQL server
+          type: string
+        xlog_segment_size:
+          description: >
+            The current WAL segment size in bytes as reported by the PostgreSQL server
+          type: integer
+      required:
+        - archive_command
+        - archive_mode
+
+    ReplicationSlot:
+      # In OpenAPI 3.1 we can use prefixItems here to be more specific however with
+      # OpenAPI 3.0 we must do the best we can...
+      description: >
+        Details of the replication slot being used by Barman on this PostgreSQL server
+        in the form of an ordered array containing the slot_name, active and
+        restart_lsn fields from pg_replication_slots
+      type: array
+      items:
+        oneOf:
+          - description: The name of the replication slot
+            type: string
+          - description: Whether the replication slot is being actively used
+            type: boolean
+          - description: >
+              The LSN of the oldest WAL which still might be required by the consumer
+              of this slot
+            type: string
+      maxItems: 3
+      minItems: 3
+
+    Wals:
+      type: object
+      properties:
+        last_archived_wal_per_timeline:
+          description: Details of the most recently archived WAL on the Barman server
+          type: object
+          properties:
+            # TIMELINE is a placeholder in lieu of patternProperties
+            TIMELINE:
```

Comment:
The BACKUP_ID placeholder was in double quotes on line 422. Is there a reason this one doesn't? (Possibly it doesn't even matter; my google fu is failing me finding the docs about placeholders like this.)

Reply:
I don't think there's anything wrong with your google-fu - this was my (bad) attempt at dealing with the lack of patternProperties and isn't valid OpenAPI (or even a common convention).

A better approach might be to use [additionalProperties](http://json-schema.org/understanding-json-schema/reference/object.html#additional-properties) since these are allowed in 3.0.0. We don't get any validation that the keys are backup IDs or timelines but we can at least say "any key names but the value must match this schema" which is an improvement.